presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -115,6 +118,20 @@ public Session getSession()
         return session;
     }
 
+    public long getPeakMemoryInBytes()
+    {
+        return peakMemory.get();
+    }
+
+    public void updateMemoryUsage(long deltaMemoryInBytes)
+    {
+        currentMemory.addAndGet(deltaMemoryInBytes);
```

Comment:
This still has a race in it. What is `peakMemory` changes between the calls to `get()` and `getAndSet()`?